src/components/base/OakCloudinaryImage/cloudinary.ts

```diff
@@ -0,0 +1,34 @@
+import { getCldImageUrl, type CldImageProps } from "next-cloudinary";
+
+let cloudinaryConfig: CldImageProps["config"] = undefined;
+
+export function setCloudinaryConfig(config: CldImageProps["config"]) {
+  cloudinaryConfig = Object.freeze({ ...config });
+}
+
+export function getCloudinaryConfig() {
+  if (!cloudinaryConfig) {
+    throw new Error(
+      "Cloudinary config is not set for `OakCloudinaryImage`, use `setCloudinaryConfig` to set it.",
+    );
+  }
+  return cloudinaryConfig;
+}
```

Comment:
🚿 thoughts from last night. Coming back to this today and this should definitely be a provider. I'll make it so.